### PR TITLE
Endpoints to edit mood and insulin data

### PR DESCRIPTION
**Pull request by RinseV**
Routed via ``/mood?modify=true`` to differentiate between adding new data vs modifying existing data. Insulin is routed via ``/insulin``. Both are simple POST endpoints that _should_ only require a JSON body with the relevant model fields after which they'll be uploaded to GameBus.

Simple tests are included, though nothing very rigorous, but I don't think these endpoints will be used.

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -19,6 +20,15 @@ if (process.env.PORT) {
     port = Number(process.env.PORT);
 }
 
+// Needed for POST requests JSON body
+// Ignore the deprecated warning: https://github.com/expressjs/body-parser/issues/428
+app.use(
```
Comment (RikLitjens) [inline src/server.ts:25]: since you linked the issue it should not become a problem

File: test/routes.test.ts [hunk 2]
```diff
@@ -79,9 +80,32 @@ describe('mood endpoint', () => {
             arousal: 1,
             valence: 1
         };
-        // TODO: route not working
         const response = await request(server).post('/mood').send(moodData);
-        expect(response.statusCode).toBe(404);
+        expect(response.statusCode).toBe(401);
+    });
+
+    test('PUT mood data', async () => {
```
Comment (RikLitjens) [inline test/routes.test.ts:87]: I had the same problem with my tests, seems like the only solution to get some coverage with these is indeed the 401.

**Review by RikLitjens — APPROVED** [review]

File: src/routes/insulin.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+insulinRouter.post('/insulin', checkJwt, async (req: any, res: Response) => {
+    // Simple POST should only include an InsulinModel
+
+    const insulinTime = req.body.timestamp as number;
+    const insulinAmount = req.body.insulinAmount as number;
+    const insulinType = req.body.insulinType as InsulinType;
+    const activityId = req.body.activityId as number;
+
+    if (!validUnixTimestamp(insulinTime) || insulinAmount < 0 || !activityId) {
```
Comment (RoelMK) [inline src/routes/insulin.ts:19]: insulinType defaults to InsulinType.RAPID if req.body.insulinType is undefined, which is unclear from this code without looking add InsulinType. Maybe add an extra check to force insulinType to be defined or else respond with status 400.

File: src/routes/mood.ts [hunk 4]
```diff
@@ -31,9 +36,47 @@ moodRouter.post('/mood', (req: any, res: Response) => {
...
+        }
+        // Get activity ID
+        const activityId = req.body.activityId as number;
+        moodModel = {
+            ...moodModel,
+            activityId: activityId
+        };
+        try {
+            gbClient.mood().putSingleMoodActivity(moodModel, req.user.playerId);
+            res.sendStatus(201).send(moodModel);
```
Comment (RoelMK) [inline src/routes/mood.ts:53]: Should this not return? Now the POST part is also executed after the PUT.

File: src/server.ts (hunk 1 above)
Comment (RoelMK) [inline src/server.ts:25]: Express.json should already do what bodyparser.json does (https://stackoverflow.com/questions/47232187/express-json-vs-bodyparser-json)

**Review by RoelMK — CHANGES_REQUESTED** [review]

File: src/routes/insulin.ts (hunk 3 above)
Comment (RinseV) [reply src/routes/insulin.ts]: Added

**Review by RoelMK — APPROVED** [review]

**Pull request merged**